audit-checklists/Wormhole.md

```diff
@@ -0,0 +1,672 @@
+# Wormhole Security Checklist
+
+## Overview
+
+Wormhole is a cross-chain messaging protocol that provides several services:
+- Message passing
+- Fungible token bridging
+- NFT bridging
+- Native token transfers (NTT)
+- Cross-chain governance
+
+The protocol operates as a multisig bridge system with the following key components:
+- Guardian nodes: A minimum of 13 guardian nodes must attest to messages (VAA's) for them to be valid
+- Relayers: Trustless parties responsible for delivering VAAs to destination chains
+- Wormhole-Core contract: Verifies VAAs on destination chains
+
+## Core Security Assumptions
+
+1. Bridge security relies on the guardian node committee
+2. VAAs are broadcast publicly and can be verified on any chain with Wormhole-Core
+3. Relayers are untrusted and may drop or skip VAAs
+4. For protocols deployed on multiple chains, implement replay protection
+
+## Service-Specific Security Considerations
+
+Services provided by Wormhole and their security issues you should consider. 
+
+1. Message Passing 
+2. Token Bridge
+3. Native Token Transfers
+4. CCTP
+5. Relayers
+
+### Direct `publishMessage` Usage
+
+Protocols can publish messages directly through the Wormhole-Core contract's `publishMessage` function:
+
+```solidity
+function publishMessage(
+    uint32 nonce,
+    bytes memory payload,
+    uint8 consistencyLevel
+) external payable returns (uint64 sequence);
+```
```

Comment:
See comment above you can remove this code snippet since you have the link to the function. It makes sense to post code snippets if you want to highlight a specific line in the function to understand the flow easier.